src/hooks/usePreferences.js

```diff
@@ -1,38 +1,53 @@
-import { useCallback, useRef } from 'react'
+import { useRef } from 'react'
 import { useHistory, useLocation } from 'react-router-dom'
-import { getPreferencesSearch } from '../Routes'
+
+// Preferences base query string
+const GLOBAL_PREFERENCES_QUERY_PARAM = '?preferences='
+
+export function getPreferencesSearch(screen) {
+  return `${GLOBAL_PREFERENCES_QUERY_PARAM}${screen}`
+}
+
+function parsePreferences(search = '') {
+  const [, path = ''] = search.split(GLOBAL_PREFERENCES_QUERY_PARAM)
```

Comment:
We may want to use [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to do this parsing instead—I'm always a bit queasy doing a naive string match on `?<param>=` 😄